frontend/src/pages/UserHome.js

```diff
@@ -1,10 +1,117 @@
+import Navbar from "../components/navbar";
+import usePagination from "../hooks/usePagination";
+import { useState } from "react";
+import Pagination from "../components/pagination";
+import Moment from "react-moment";
+import useGetActivities from "../hooks/useGetActivities";
+import { ArrowPathIcon } from "@heroicons/react/24/solid";
+import { ExclamationTriangleIcon } from "@heroicons/react/24/outline";
+import useGetUserInfo from "../hooks/useGetUserInfo";
+import useGetLearningsCount from "../hooks/useGetLearningsCount";
+
 const UserHome = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+
+  const {
+    error: acitvityError,
+    isLoading: activityLoad,
+    data: activityDadta,
+  } = useGetActivities("null");
+
+  const {
+    error: userInfoError,
+    isLoading: UserInfoLoad,
+    data: userInfo,
+  } = useGetUserInfo("null");
+
+  const {
+    error: learningsError,
+    isLoading: learningsLoad,
+    data: learntWordsAndLessons,
+  } = useGetLearningsCount("null");
+
+  const { handlePageClick, currentItems, pageCount } = usePagination(
+    itemsPerPage,
+    activityDadta ? activityDadta : []
+  );
   return (
-    <div>
-      <h1 className="text-[3rem] font-bold">This is my homepage</h1>
-      <h2 className="text">
-        Please note that currently there is no route protection
-      </h2>
+    <div className="min-h-[100vh] w-[100%] h-[100%] flex flex-col">
+      <Navbar />
+      <main className=" flex-grow flex items-center justify-center ">
+        <div className="flex sm:flex-col sm:space-x-0 sm:space-y-4 lg:space-y-0 sm:items-center lg:flex-row lg:space-x-24 w-[70%] justify-center lg:items-start m-3">
+          {/* Right side */}
+          <div className="inline bg-white p-6 rounded-xl shadow-md min-w-fit space-y-4">
+            <h2 className="text-[2.5rem] font-medium border-b-2">Dashboard</h2>
```

Comment:
- code here is bit messy. I believe you can still restructure and split the components by pieces.

example in the image:
![image](https://user-images.githubusercontent.com/73863685/188371994-dfba4347-4c1e-4362-b622-94a528aaeb55.png)

```
/pages
  /Home
   Home.js
   /components
     Navbar.js
     UserInfo.js
     Activities.js
```